src/Adapter/Module/ModuleZipManager.php

```diff
@@ -0,0 +1,214 @@
+<?php
+/*
+ * 2007-2016 PrestaShop
+ * 
+ * NOTICE OF LICENSE
+ * 
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ * 
+ * DISCLAIMER
+ * 
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ * 
+ *  @author PrestaShop SA <contact@prestashop.com>
+ *  @copyright  2007-2016 PrestaShop SA
+ *  @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *  International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Module;
+
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Finder\Finder;
+use Symfony\Component\Translation\TranslatorInterface;
+
+use Exception;
+use Tools;
+
+class ModuleZipManager
+{
+    /*
+     * Data
+     */
+    private static $sources = array();
+    private $attributes = array('name', 'sandboxPath');
+
+    /*
+     * Services
+     */
+    /**
+     * @var Symfony\Component\Filesystem\Filesystem
+     */
+    private $filesystem;
+    /**
+     * @var Symfony\Component\Finder\Finder
+     */
+    private $finder;
+    private $translator;
+    
+    public function __construct(TranslatorInterface $translator)
```

Comment:
can you put all dependencies as arguments of constructor ?

pros: easier to test.
cons: no cons ?